Unable to get gamestate for game in progress

Seen on main as of [19b2a61](https://github.com/trekcc-ops/gemp-stccg/commit/19b2a611dfef976b5957149d42edf930b2ed824c). When I have a game of 1E in progress and I try to go to the URL for its gamestate, http://localhost:17001/gemp-stccg-server/game/1/gameState/player , then the server throws exceptions with the following errors in the log:

```
gemp_app_1  | Error response for /gemp-stccg-server/game/1/gameState/player
gemp_app_1  | java.lang.IllegalArgumentException: Could not resolve type id 'game' as a subtype of `com.gempukku.stccg.async.handler.UriRequestHandlerNew`: known type ids = [addSealedLeague, banUser, banUserTemporary, cancelGame, clearCache, concedeGame, createTable, currentTournaments, decisionResponse, deckFormats, deckStats, deleteDeck, gameCardInfo, gameHistory, getChat, getErrata, getGameState, getHall, getSets, importDeck, joinQueue, joinTable, leaveQueue, leaveTable, leaveTournament, listLibraryDecks, listUserDecks, login, playerInfo, playerStats, postChat, previewSealedLeague, register, reloadCardLibrary, renameDeck, replay, saveDeck, sendChatMessage, serverStats, setShutdown, startGameSession, tournamentHistory, unBanUser, updateGameState, updateHall]
gemp_app_1  |  at [Source: UNKNOWN; byte offset: #UNKNOWN]
gemp_app_1  |   at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:4624) ~[web.jar:?]
gemp_app_1  |   at com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:4555) ~[web.jar:?]
gemp_app_1  |   at com.gempukku.stccg.async.GempukkuHttpRequestHandler.handleRequest(GempukkuHttpRequestHandler.java:111) ~[web.jar:?]
gemp_app_1  |   at com.gempukku.stccg.async.GempukkuHttpRequestHandler.channelRead0(GempukkuHttpRequestHandler.java:56) ~[web.jar:?]
gemp_app_1  |   at com.gempukku.stccg.async.GempukkuHttpRequestHandler.channelRead0(GempukkuHttpRequestHandler.java:1) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[web.jar:?]
gemp_app_1  |   at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[web.jar:?]
gemp_app_1  |   at io.netty.handler.codec.MessageToMessageCodec.channelRead(MessageToMessageCodec.java:111) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[web.jar:?]
gemp_app_1  |   at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436) ~[web.jar:?]
gemp_app_1  |   at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346) ~[web.jar:?]
gemp_app_1  |   at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1357) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:868) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650) ~[web.jar:?]
gemp_app_1  |   at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) ~[web.jar:?]
gemp_app_1  |   at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[web.jar:?]
gemp_app_1  |   at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[web.jar:?]
gemp_app_1  |   at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[web.jar:?]
gemp_app_1  |   at java.lang.Thread.run(Thread.java:1583) [?:?]
gemp_app_1  | Caused by: com.fasterxml.jackson.databind.exc.InvalidTypeIdException: Could not resolve type id 'game' as a subtype of `com.gempukku.stccg.async.handler.UriRequestHandlerNew`: known type ids = [addSealedLeague, banUser, banUserTemporary, cancelGame, clearCache, concedeGame, createTable, currentTournaments, decisionResponse, deckFormats, deckStats, deleteDeck, gameCardInfo, gameHistory, getChat, getErrata, getGameState, getHall, getSets, importDeck, joinQueue, joinTable, leaveQueue, leaveTable, leaveTournament, listLibraryDecks, listUserDecks, login, playerInfo, playerStats, postChat, previewSealedLeague, register, reloadCardLibrary, renameDeck, replay, saveDeck, sendChatMessage, serverStats, setShutdown, startGameSession, tournamentHistory, unBanUser, updateGameState, updateHall]
gemp_app_1  |  at [Source: UNKNOWN; byte offset: #UNKNOWN]
gemp_app_1  |   at com.fasterxml.jackson.databind.exc.InvalidTypeIdException.from(InvalidTypeIdException.java:43) ~[web.jar:?]
gemp_app_1  |   at com.fasterxml.jackson.databind.DeserializationContext.invalidTypeIdException(DeserializationContext.java:2040) ~[web.jar:?]
gemp_app_1  |   at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownTypeId(DeserializationContext.java:1590) ~[web.jar:?]
gemp_app_1  |   at com.fasterxml.jackson.databind.jsontype.impl.TypeDeserializerBase._handleUnknownTypeId(TypeDeserializerBase.java:298) ~[web.jar:?]
gemp_app_1  |   at com.fasterxml.jackson.databind.jsontype.impl.TypeDeserializerBase._findDeserializer(TypeDeserializerBase.java:165) ~[web.jar:?]
gemp_app_1  |   at com.fasterxml.jackson.databind.jsontype.impl.AsPropertyTypeDeserializer._deserializeTypedForId(AsPropertyTypeDeserializer.java:151) ~[web.jar:?]
gemp_app_1  |   at com.fasterxml.jackson.databind.jsontype.impl.AsPropertyTypeDeserializer.deserializeTypedFromObject(AsPropertyTypeDeserializer.java:136) ~[web.jar:?]
gemp_app_1  |   at com.fasterxml.jackson.databind.deser.AbstractDeserializer.deserializeWithType(AbstractDeserializer.java:263) ~[web.jar:?]
gemp_app_1  |   at com.fasterxml.jackson.databind.deser.impl.TypeWrappedDeserializer.deserialize(TypeWrappedDeserializer.java:74) ~[web.jar:?]
gemp_app_1  |   at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:4619) ~[web.jar:?]
gemp_app_1  |   ... 37 more
```

--- a/gemp-module/gemp-stccg-server/src/main/java/com/gempukku/stccg/async/GempukkuHttpRequestHandler.java
+++ b/gemp-module/gemp-stccg-server/src/main/java/com/gempukku/stccg/async/GempukkuHttpRequestHandler.java
@@ -2,6 +2,7 @@ package com.gempukku.stccg.async;
 
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.exc.InvalidTypeIdException;
 import com.gempukku.stccg.async.handler.*;
 import com.gempukku.stccg.common.AppConfig;
 import io.netty.channel.ChannelHandlerContext;
@@ -85,33 +86,43 @@ public class GempukkuHttpRequestHandler extends SimpleChannelInboundHandler<Full
             }
             boolean requestHandled = false;
 
-            String afterServer = uri.substring(SERVER_CONTEXT_PATH.length());
-            int nextSlashIndex = afterServer.indexOf("/");
-            String handlerType = (nextSlashIndex < 0) ? afterServer : afterServer.substring(0, nextSlashIndex);
-            String afterHandlerType = afterServer.substring(handlerType.length());
+            if (uri.startsWith(SERVER_CONTEXT_PATH)) {
 
-            UriRequestHandler handler = switch(handlerType) {
-                case "admin" -> new AdminRequestHandler(_serverObjects);
-                case "collection" -> new CollectionRequestHandler(_serverObjects);
-                case "league" -> new LeagueRequestHandler(_serverObjects);
-                case "playtesting" -> new PlaytestRequestHandler(_serverObjects);
-                case "soloDraft" -> new SoloDraftRequestHandler(_serverObjects);
-                case "tournament" -> new TournamentRequestHandler(_serverObjects);
-                default -> null;
-            };
-            if (handler != null) {
-                handler.handleRequest(afterHandlerType, request, responseWriter);
-                requestHandled = true;
-            }
+                String afterServer = uri.substring(SERVER_CONTEXT_PATH.length());
+                int nextSlashIndex = afterServer.indexOf("/");
+                String handlerType = (nextSlashIndex < 0) ? afterServer : afterServer.substring(0, nextSlashIndex);
+                String afterHandlerType = afterServer.substring(handlerType.length());
 
-            if (!requestHandled) {
-                Map<String, String> parameters = request.parameters();
-                parameters.put("type", handlerType);
-                try {
-                    UriRequestHandlerNew newHandler = _jsonMapper.convertValue(parameters, UriRequestHandlerNew.class);
-                    newHandler.handleRequest(request, responseWriter, _serverObjects);
-                } catch(JsonProcessingException exp) {
-                    throw new HttpProcessingException(HttpURLConnection.HTTP_NOT_FOUND); // 404
+                UriRequestHandler handler = switch (handlerType) {
+                    case "admin" -> new AdminRequestHandler(_serverObjects);
+                    case "collection" -> new CollectionRequestHandler(_serverObjects);
+                    case "league" -> new LeagueRequestHandler(_serverObjects);
+                    case "playtesting" -> new PlaytestRequestHandler(_serverObjects);
+                    case "soloDraft" -> new SoloDraftRequestHandler(_serverObjects);
+                    case "tournament" -> new TournamentRequestHandler(_serverObjects);
+                    default -> null;
+                };
+                if (handler != null) {
+                    handler.handleRequest(afterHandlerType, request, responseWriter);
+                    requestHandled = true;
+                }
+
+                if (!requestHandled) {
+                    Map<String, String> parameters = request.parameters();
+                    parameters.put("type", handlerType);
+                    try {
+                        UriRequestHandlerNew newHandler =
+                                _jsonMapper.convertValue(parameters, UriRequestHandlerNew.class);
+                        newHandler.handleRequest(request, responseWriter, _serverObjects);
+                    } catch (IllegalArgumentException exp) {
+                        if (exp.getCause() instanceof InvalidTypeIdException) {
+                            // InvalidTypeIdException thrown if initial path not recognized by the Json deserializer
+                            logHttpError(uri,
+                                    new HttpProcessingException(HttpURLConnection.HTTP_NOT_FOUND), responseWriter);
+                        }
+                    } catch (JsonProcessingException exp) {
+                        throw new HttpProcessingException(HttpURLConnection.HTTP_NOT_FOUND); // 404
+                    }
                 }
             }
         }

